hooks/shapediver/viewer/interaction/utils/patternUtils.ts

```diff
@@ -0,0 +1,92 @@
+import { notifications } from "@mantine/notifications";
+import { IInteractionParameterSettings, ISessionApi, ITreeNode, OutputApiData } from "@shapediver/viewer";
+
+/**
+ * Process the nameFilter and update the pattern.
+ * 
+ * @param sessionApi The session API used to get output IDs by name.
+ * @param settings The settings object containing nameFilter.
+ * @returns The pattern object with updated patterns.
+ */
+export const processPattern = (sessionApi: ISessionApi, settings?: IInteractionParameterSettings): { [key: string]: string[][] } => {
+	const pattern: {
+		[key: string]: string[][];
+	} = {};
+
+	let nameFilter: string[] = [];
+
+	if (settings && settings.props.nameFilter !== undefined) {
+		if (typeof settings.props.nameFilter === "string") {
+			if ((settings.props.nameFilter as string).startsWith("[") && (settings.props.nameFilter as string).endsWith("]") && (settings.props.nameFilter as string) !== "[]") {
+				try {
+					nameFilter = JSON.parse(settings.props.nameFilter) as string[];
+				} catch (e) {
+					notifications.show({
+						title: "Invalid Name Filter",
+						message: "The name filter is not a valid JSON array."
+					});
+				}
```

Comment:
@MajorMeerkatThe3rd this should never happen, let's add validation where we receive the data (e.g. in appbuildertypecheck.ts)